Cutelyst/application.cpp

```diff
@@ -441,6 +442,96 @@ bool Application::enginePostFork()
     return true;
 }
 
+void Application::addTranslator(const QLocale &locale, QTranslator *translator)
+{
+    Q_D(Application);
+    Q_ASSERT_X(translator, "add translator to application", "invalid QTranslator object");
+    if (d->translators.contains(locale)) {
+        QVector<QTranslator*> trs = d->translators.value(locale);
+        trs.prepend(translator);
+        d->translators.insert(locale, trs);
+    } else {
+        d->translators.insert(locale, QVector<QTranslator*>(1, translator));
+    }
+}
+
+void Application::addTranslator(const QString &locale, QTranslator *translator)
+{
+    addTranslator(QLocale(locale), translator);
+}
+
+void Application::addTranslators(const QLocale &locale, const QVector<QTranslator *> &translators)
+{
+    Q_D(Application);
+    Q_ASSERT_X(!translators.empty(), "add translators to application", "empty translators vector");
+    if (d->translators.contains(locale)) {
+        QVector<QTranslator*> trs = d->translators.value(locale);
+        for (auto it = translators.crbegin(); it != translators.crend(); ++it) {
+            trs.prepend(*it);
+        }
+        d->translators.insert(locale, trs);
+    } else {
+        d->translators.insert(locale, translators);
+    }
+}
+
+static void replacePercentN(QString *result, int n)
```

Comment:
Is this function really needed? I thought substitution would be done automatically, and identation
 is different